specification/k3po.control/SPEC.md

```diff
@@ -0,0 +1,195 @@
+sends back a STARTED event. 
+
+###### Headers
+N/A
+
+###### Content
+N/A
+
+
+##### ABORT command
+
+###### Description
+This command will abort the current execution. It can be sent also after a PREPARE command (and before a START command).
+TO DO - not sure what is sent back after an ABORT.
+
+###### Headers
+N/A
+
+###### Content
+N/A
+
+
+##### AWAIT command
+
+###### Description
+This command will respond with a NOTIFIED event after the barrier with the name sent in the header is triggered on the k3po server.
+
+###### Headers
+- barrier - the name of the barrier that should be waited on
+
+###### Content
+N/A
+
+
+##### NOTIFIED command
+
+###### Description
+This command will trigger the barrier with the name sent in the header and will respond with a NOTIFIED event with the same barrier name.
+
+###### Headers
+- barrier - the name of the barrier that should be triggered
+
+###### Content
+N/A
+
+
+##### DISPOSE command
+
+###### Description
+This command will notify the k3po server that it should end the session and clean-up the resources. When execution is finished, a DISPOSED
+event is sent back.
+
+###### Headers
+N/A
+
+###### Content
+N/A
+
+
+### EVENTS
+
+##### PREPARED event
+
+###### Description
+This event is sent by the server after a PREPARE command executed successfully.
+
+###### Headers
+- content-length - the length of the content of the command. This header is optional, should be present only when content is not empty.
+- barrier - the name of a barrier in the prepared script. Can occur multiple times (multiple barriers).
+	
+###### Content
+The content for prepared event is the content of the script that will be executed on the k3po server. 
+
+
+
+##### STARTED event
+
+###### Description
+This event is sent by the server after a START command executed successfully.
+
+###### Headers
+N/A
+	
+###### Content
+N/A 
+
+
+##### NOTIFIED event
+
+###### Description
+This event is sent by the server when a barrier is triggered.
+
+###### Headers
+- barrier - the name of the barrier that was triggered.
+	
+###### Content
+N/A 
+
+
+##### DISPOSED event
+
+###### Description
+This event is sent by the server when the server was cleaned-up. It is sent normally as a response for a DISPOSE command.
+
+###### Headers
+N/A
+	
+###### Content
+N/A 
+
+
+##### FINISHED event
+
+###### Description
+This event is sent by the server when the server has finished execution of the scripts.
+
+###### Headers
+- content-length - the length of the content of the command. This header is optional, should be present only when content is not empty.
+	
+###### Content
+The content for the event is the observed script after the execution on the k3po server. This script can be compared with the script in the
+prepared event to check if the test was successful. 
+
+
+##### ERROR event
+
+###### Description
+This event is sent by the server when the server encountered an error while processing another command.
```

Comment:
Currently, the client is sending a DISPOSE after an error event is received.

I think the server does nothing, I'm not sure. It remains in the same state as before, after sending the error event.